File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& arr, int k) {
        unordered_map<int,int> freq;
        int sum = 0;
        int count = 0;
        freq[0] = 1;

        for(int x : arr) {
            sum += x;
            if(freq.find(sum - k) != freq.end()) {
                count += freq[sum - k];
            }
            freq[sum]++;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, arr, k):
        n=len(arr)
        freq={0:1}
        sum=0
        count=0
        for x in arr:
            sum+=x
            if sum-k in freq:
                count+=freq[sum-k]
            freq[sum]=freq.get(sum,0)+1
        return count